app/src/main/java/org/kiwix/kiwixmobile/local_file_transfer/LocalFileTransferActivity.kt

```diff
@@ -0,0 +1,414 @@
+          )
+          showToast(
+            this,
+            R.string.permission_refused_location,
+            Toast.LENGTH_LONG
+          )
+          finish()
+        }
+        PERMISSION_REQUEST_CODE_STORAGE_WRITE_ACCESS -> {
+          Log.e(
+            TAG,
+            "Storage write permission not granted"
+          )
+          showToast(
+            this,
+            R.string.permission_refused_storage,
+            Toast.LENGTH_LONG
+          )
+          finish()
+        }
+        else -> {
+          super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        }
+      }
+    }
+  }
+
+  private val isLocationServiceEnabled: Boolean
+    get() = (isProviderEnabled(LocationManager.GPS_PROVIDER)
+      || isProviderEnabled(LocationManager.NETWORK_PROVIDER))
+
+  private fun isProviderEnabled(locationProvider: String): Boolean {
+    return try {
+      locationManager!!.isProviderEnabled(locationProvider)
+    } catch (ex: SecurityException) {
+      ex.printStackTrace()
+      false
+    } catch (ex: IllegalArgumentException) {
+      ex.printStackTrace()
+      false
+    }
+  }
+
+  private fun requestEnableLocationServices() {
+    alertDialogShower!!.show(KiwixDialog.EnableLocationServices, {
+      startActivityForResult(
+        Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS),
+        REQUEST_ENABLE_LOCATION_SERVICES
+      )
+    }, {
+      showToast(
+        this@LocalFileTransferActivity, R.string.discovery_needs_location,
+        Toast.LENGTH_SHORT
+      )
+    })
+  }
+
+  private fun requestEnableWifiP2pServices() {
+    alertDialogShower!!.show(KiwixDialog.EnableWifiP2pServices, {
+      startActivity(Intent(Settings.ACTION_WIFI_SETTINGS))
+    }, {
+      showToast(
+        this@LocalFileTransferActivity, R.string.discovery_needs_wifi,
+        Toast.LENGTH_SHORT
+      )
+    })
+  }
+
+  override fun onActivityResult(
+    requestCode: Int,
+    resultCode: Int,
+    data: Intent?
+  ) {
+    when (requestCode) {
+      REQUEST_ENABLE_LOCATION_SERVICES -> {
+        if (!isLocationServiceEnabled) {
+          showToast(
+            this,
+            R.string.permission_refused_location,
+            Toast.LENGTH_LONG
+          )
+        }
+      }
+      else -> {
+        super.onActivityResult(requestCode, resultCode, data)
+      }
+    }
+  }
+
+  override fun onDestroy() {
+    wifiDirectManager!!.stopWifiDirectManager()
+    super.onDestroy()
+  }
+
+  override fun injection(coreComponent: CoreComponent) {
+    this.kiwixActivityComponent.inject(this)
+  }
+
+  companion object {
+    // Not a typo, 'Log' tags have a length upper limit of 25 characters
+    const val TAG = "LocalFileTransferActvty"
+    const val REQUEST_ENABLE_LOCATION_SERVICES = 1
+    private const val PERMISSION_REQUEST_CODE_COARSE_LOCATION = 1
+    private const val PERMISSION_REQUEST_CODE_STORAGE_WRITE_ACCESS = 2
+
+    /* Miscellaneous helper methods */
+    fun showToast(
```

Comment:
get rid of these showToast methods, now that we are in kotlin we have access to our Activitiy extension function `toast`